### PR TITLE
ci: Move MinGW jobs to Github actions

### PR DESCRIPTION
**Pull request by sfan5**
not sure if we want this but it's finished and works so here's the PR
(alternative to #8817 essentially)

@JEEB

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name: build
+
+on:
+  push:
+    branches:
+      - master
+      - ci
+      - 'release/**'
```
Comment (jeeb) [inline .github/workflows/build.yml:8]: Probably we want to also do these runs on PRs as well, see https://github.com/obsproject/obs-studio/actions/runs/830902393/workflow#L3-L15 as an example.

File: TOOLS/travis-rebuild-website [hunk 2]
```diff
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-if [ "x$TARGET" != "xx86_64-w64-mingw32" ]; then
+if [[ "$TRAVIS_OS_NAME" != "linux" || "$CC" != "clang" ]]; then
```
Comment (jeeb) [inline TOOLS/travis-rebuild-website:3]: It feels like something is not exactly right here looking at the job log:
```
$ TOOLS/travis-rebuild-website
TOOLS/travis-rebuild-website: [[: not found
clang: error: no such file or directory: '!='
clang: error: no such file or directory: 'clang'
clang: error: no such file or directory: ']]'
clang: error: no input files
```

File: .github/workflows/build.yml (hunk 1 above)
Comment (sfan5) [reply .github/workflows/build.yml:8]: Thought it'd do that automatically if I don't specify but I guess not.

**Review by jeeb — APPROVED** [review]
LGTM.

Seems like Ubuntu removed the broken 9.3 packages, so most likely if Travis would regenerate their image it'd get fixed but it still boggles the mind how a funky libc update could get so far...

```
     Downgrade the following packages:
1)     libc-dev-bin [2.31-0ubuntu9.3 (now) -> 2.31-0ubuntu9.2 (focal-updates
2)     libc6 [2.31-0ubuntu9.3 (now) -> 2.31-0ubuntu9.2 (focal-updates)]
3)     libc6-dbg [2.31-0ubuntu9.3 (now) -> 2.31-0ubuntu9.2 (focal-updates)]
4)     libc6-dev [2.31-0ubuntu9.3 (now) -> 2.31-0ubuntu9.2 (focal-updates)]
5)     libc6-i386 [2.31-0ubuntu9.3 (now) -> 2.31-0ubuntu9.2 (focal-updates)]
```

**Pull request merged**